### ISSUE TITLE
How to retrain the existing initial model.

### ISSUE THREAD

**Issue by arunnbaba** [talmago/spacy_crfsuite#4]
Hi,

Thanks for your clarification on the issue which I have raised later. It helped me a lot.

Could you please mention how to train an existing mode. say, I have trained the model with 1000 data and to enhance the model accuracy further I want to train the model on another 500 data points. How to do this?.

**Comment by talmago** [comment]
@arunnbaba I'm not sure that i fully understand the term "train an existing mode". 

But i'll try to describe the process i would do to train an entity tagger. I'd start by splitting the gold data into 800 training examples (say `train.md`) and 200 dev examples (`dev.md`). I'd later try to fine-tune the "splitting" method in order to equally spread the gold labels for train/dev. 

then, i'd train on the training set. 

```
$ python -m spacy_crfsuite.train train.md -o model/ [-c config.json]
```

And evaluate the baseline F1 score on the dev set.

```
$ python -m spacy_crfsuite.eval dev.md -m model/model.pkl
```

Now it's a good time to fine-tune the model hyper-params. Currently, and unfortunately i must say, it's only possible to fine-tune them on the dev set. This is done by benchmarking the above process with several config files. However, i'm planning to release a grid-search functionality for training, which will cross-validate the hyper-params on the complete set.

Given the analysis of your model, you can choose to add more training examples. In this case, i'd recommend to create a 2nd version of the dataset split, and re-iterate.

Hope it helps !

**Comment by arunnbaba** [comment]
Hi talmago,

Thanks for your quick reply. What am asking is, You have trained the model with 800 data points(train.md ) and the model is ready for production. If I want to again train the model with more data, how to accomplish that?

**Comment by talmago** [comment]
@arunnbaba so currently there is no way to train the model from a checkpoint, so you will need to run the `train` command again for the new train set and output location. 

As for training from a checkpoint (a pre-built model) - i'll take a look at that and see what can be done. But i must say, for a small dataset, it should barely make a difference.

**Comment by arunnbaba** [comment]
Thank you

**Comment by talmago** [comment]
@arunnbaba I just released a new version (1.0.2) which enables also training from a checkpoint.

So you can resume training with additional examples by running - 

```
$ python -m spacy_crfsuite.train examples/new_examples.md -o model2/ -c config.json -m model/model.pkl
```

Make sure to wipe down your pipenv/conda environment when re-installing the spacy_crfsuite package.
